packages/syft/src/syft/client/domain_client.py

```diff
@@ -88,28 +91,32 @@ def upload_dataset(self, dataset: CreateDataset) -> Union[SyftSuccess, SyftError
         # relative
         from ..types.twin_object import TwinObject
 
+        if self.users is None:
+            return SyftError(f"can't get user service for {self}")
+
         user = self.users.get_current_user()
         dataset = add_default_uploader(user, dataset)
         for i in range(len(dataset.asset_list)):
             asset = dataset.asset_list[i]
             dataset.asset_list[i] = add_default_uploader(user, asset)
 
         dataset._check_asset_must_contain_mock()
-        dataset_size = 0
+        dataset_size: float = 0.0
 
         # TODO: Refactor so that object can also be passed to generate warnings
-        metadata = self.api.connection.get_node_metadata(self.api.signing_key)
-
-        if (
-            metadata.show_warnings
-            and metadata.node_side_type == NodeSideType.HIGH_SIDE.value
-        ):
-            message = (
-                "You're approving a request on "
-                f"{metadata.node_side_type} side {metadata.node_type} "
-                "which may host datasets with private information."
-            )
-            prompt_warning_message(message=message, confirm=True)
+        if self.api.connection:
```

Comment:
Let's cast API over here. 
`self.api = cast(SyftAPI, self.api)`

Reply:
But Shubham here `SyftAPI.connection` can still be None, so that's why we check it there

Reply:
Type casted to NodeConnection: 0596219b8584c888c056f9084e0f590bc1075e7a